Backoffice/EventSubscriber/UpdateChildNodePathSubscriber.php

```diff
@@ -21,33 +20,34 @@ class UpdateChildNodePathSubscriber implements EventSubscriberInterface
     /**
      * @param NodeRepositoryInterface  $nodeRepository
      * @param EventDispatcherInterface $eventDispatcher
-     * @param CurrentSiteIdInterface   $currentSiteManager
      */
-    public function __construct(NodeRepositoryInterface $nodeRepository, EventDispatcherInterface $eventDispatcher, CurrentSiteIdInterface $currentSiteManager)
+    public function __construct(NodeRepositoryInterface $nodeRepository, EventDispatcherInterface $eventDispatcher)
     {
         $this->nodeRepository = $nodeRepository;
         $this->eventDispatcher = $eventDispatcher;
-        $this->currentSiteManager = $currentSiteManager;
     }
 
     /**
      * @param NodeEvent $event
      */
     public function updatePath(NodeEvent $event)
     {
-        $node = $event->getNode();
-        $parentPath = $node->getPath();
-        $siteId = $this->currentSiteManager->getCurrentSiteId();
-        $sons = $this->nodeRepository->findByParent($node->getNodeId(), $siteId);
+        $parentNode = $event->getNode();
+        $events = array();
+        $nodesVersions = $this->nodeRepository->findByParent($parentNode->getNodeId(), $parentNode->getSiteId());
 
-        $sonsToUpdate = array();
-        foreach ($sons as $son) {
-            $son->setPath($parentPath . '/' . $son->getNodeId());
-            $sonsToUpdate[$son->getNodeId()] = $son;
+        foreach ($nodesVersions as $nodeVersion) {
+            $oldPath = $nodeVersion->getPath();
+            $nodeVersion->setPath($parentNode->getPath() . '/' . $nodeVersion->getNodeId());
+
+            if (!isset($events[$nodeVersion->getNodeId()])) {
+                $event = new NodeEvent($nodeVersion);
+                $event->setPreviousPath($oldPath);
+                $events[$nodeVersion->getNodeId()] = $event;
```

Comment:
@alavieille : because only one event must be dispatched per node, not one event per node/language/version